lib/config/types.ts

```diff
@@ -368,6 +368,12 @@ export interface ValidationMessage {
   message: string;
 }
 
+export type AllowedParents =
```

Comment:
we can later add a `$root` item to indicate it's a global option. then we can later set parents for `password` to `$root,hostRule` for better validation and schema generation.